### PR TITLE
Add sconsdeps docs

### PR DESCRIPTION
**Pull request by czoido**
Docs for: https://github.com/conan-io/conan/pull/14713

### CODE REVIEW

File: reference/conanfile/tools/scons.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+.. _conan_tools_sconsdeps:
+
+conan.tools.scons
+=================
+
+.. important::
+
+   Some of the features used in this section are still **under development**. While they
+   are recommended and usable, and we will try not to break them in future releases, some
+   breaking changes might still occur.
```
Comment (franramirez688) [inline reference/conanfile/tools/scons.rst:10]: We should use this message from a common module to be included anywhere.

File: reference/conanfile/tools/scons.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+The ``SConsDeps`` is the dependency generator for `SCons <https://scons.org/>`_. It will
+generate a `SConscript_conandeps` file containing the necessary information for SCons to
+build against the desired dependencies.
+
+The ``SConsDeps`` generator can be used by name in conanfiles:
+
+.. code-block:: python
+   :caption: conanfile.py
+
+   class Pkg(ConanFile):
```
Comment (franramirez688) [inline reference/conanfile/tools/scons.rst:26]:
```suggestion
   from conan import ConanFile
 
   class Pkg(ConanFile):
```

**Review by franramirez688 — APPROVED** [review]

File: reference/conanfile/tools/scons.rst (hunk 1 above)
Comment (memsharded) [reply reference/conanfile/tools/scons.rst:10]: Yeah, not worth for 1.X

**Review by memsharded — APPROVED** [review]

File: reference/conanfile/tools/scons.rst (hunk 1 above)
Comment (franramirez688) [reply reference/conanfile/tools/scons.rst:10]: Sorry, I did not realize that it's for 1.x.

**Pull request merged**